.github/workflows/codeql.yml

```diff
@@ -0,0 +1,41 @@
+name: codeql
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
```

Comment:
Any reason for not adding it for PRs too?

Reply:
`pull_request:` means it will be triggered for any PRs against any based branch if you don't specify `branches` field under.

Reply:
Ah, right!